### PR TITLE
Add pkg/cmp/dev for I²C based SHT sensors

### PR DESCRIPTION
**Pull request by dbrgn**
Since all SHT sensors have the same pinout, we can use a single component.

The concrete devices have to wait until the DFN packages are done (@HBadertscher is working on them over at https://github.com/LibrePCB/librepcb-parts-generator/pull/22).

### CODE REVIEW

**Comment by rnestler** [comment]
> Since all SHT sensors have the same pinout, we can use a single component.

What do you mean by that? The https://www.sensirion.com/en/environmental-sensors/humidity-sensors/digital-humidity-sensors-for-various-applications/ for example comes in different variants which have different pins (I2C and analog version).

Edit: Also the SHT3x I²C sensors have an ALERT and an ADDR pin compared to the SHT2x.

**Comment by dbrgn** [comment]
> What do you mean by that?

Sorry, what I wanted to say: All I²C SHT sensors have the same pinout. (That's why the component name also includes "I²C").

However, I didn't realize that the SHT3x have additional pins...

Do you know a good way of grouping the different sensors into components? Maybe like this?

- SHT I²C 4-Pin
- SHT I²C 6-Pin
- SHT Analog 5-Pin
- SGP I²C 5-Pin

**Comment by rnestler** [comment]
> Do you know a good way of grouping the different sensors into components? Maybe like this?

I guess that makes sense. But I'm not sure what you mean by components. Isn't a component something you can order? So a SHT30-ARP-B for example?

Also since the different sensor generation have different package size I would group them by that:

 * SHT2x
 * SHT3x
 * SHT3x Analog
 * SHTCx
 * SHTWx (but there I think that even the SHTW1 has a different package)

**Comment by dbrgn** [comment]
> Isn't a component something you can order?

No, that's the device: https://developers.librepcb.org/df/d4f/doc_library.html

> Also since the different sensor generation have different package size I would group them by that:

Ok, I'll give it a try.

**Comment by dbrgn** [comment]
PR updated. It's not complete yet, but ready for merging (we can add more parts later).

img

**Comment by ubruhin** [comment]
Nice! My feedback:

Symbol:
  - I would swap nRST and VCC pins to allow placing a capacitor parallel to GND/VCC.
  - Also swap SCL/SDA pins to have consistent pin order across all SHT components

Components:
- Maybe rename "SHT3x" to "SHT3x-DIS" to make clear it's the I2C variant.

Packages: 
- I would name them according the sensor names instead of the IPC name because the "hole" on their top is an important difference to standard DFN packages. Theoretically it would be possible that two sensor types have the same footprint, but different holes on the top, then we need two separate packages with distinguishable names. Of course the IPC name can be added to the description to make clear which IPC footprint it corresponds to.
- The hand soldering footprint seem to have the documentation layer missing(?)

**Comment by dbrgn** [comment]
> The hand soldering footprint seem to have the documentation layer missing(?)

Good catch! :slight_smile:

I implemented all suggestions.

File: cmp/d847b117-0435-4c71-81aa-3f670d540e6c/component.lp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+(librepcb_component d847b117-0435-4c71-81aa-3f670d540e6c
+ (name "SHTCx")
+ (description "Digital I²C temperature / humidity sensor.\n\nInput voltage: 1.62 - 1.98 V\nTypical accuracy: ±3 %RH and ±0.3 °C\nDatasheet: https://www.sensirion.com/shtc1/")
```
Comment (ubruhin) [inline cmp/d847b117-0435-4c71-81aa-3f670d540e6c/component.lp]: The voltage range and accuracy are not valid for all SHTCx devices, thus removing them from the component description:
```suggestion
 (description "Digital I²C temperature / humidity sensor.\n\nDatasheet: https://www.sensirion.com/shtc1/")
```

**Review by ubruhin — APPROVED** [review]
See comment. Otherwise OK :+1:

File: cmp/d847b117-0435-4c71-81aa-3f670d540e6c/component.lp (hunk 1 above)
Comment (dbrgn) [reply cmp/d847b117-0435-4c71-81aa-3f670d540e6c/component.lp]: Oh, that's a leftover. The datasheet URL is also wrong. I'll fix it directly.

**Pull request merged**